### PR TITLE
[WIP] Enable pre-commit

### PR DESCRIPTION
**Pull request by yunchu**
### Summary
Enable pre-commit using the [pre-commit](pre-commit.com) tool.
* added '.pre-commit-config.yaml' to define pre-commit hooks
* fixed some issues reported by pre-commit

How to use it on your local env:
1. Install 'pre-commit' tool to your env.
`$ pip install pre-commit`
2. Install pre-commit hook configuration to your local repo
`$ pre-commit install`

Once it done above, linters will work for your changes when you perform the 'git commit' command on your env.

How to run linters to all-files in the repo
`.../datumaro $ tox -e pre-commit`

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -1,79 +1,15 @@
...
-      - uses: actions/checkout@v3
-      - name: Check whitespace
-        run: |
-          empty_tree="$(git hash-object -t tree --stdin < /dev/null)"
-          git diff --check "$empty_tree"
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run pre-commit
+        run: tox -e pre-commit
+      # - uses: pre-commit/action@v3.0.0
```
Comment (vinnamkim) [inline .github/workflows/linter.yml]: Please add the end of file and add it to the other files too.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    rev: 3.0.0
+    hooks:
+      - id: git-check  # Configure in .gitattributes
+
+  # notebooks
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.6.3
+    hooks:
+      - id: nbqa-black
+      - id: nbqa-ruff
```
Comment (vinnamkim) [inline .pre-commit-config.yaml:44]: I think it is better to add https://github.com/pre-commit/pre-commit-hooks#end-of-file-fixer

File: .github/workflows/linter.yml (hunk 1 above)
Comment (yunchu) [reply .github/workflows/linter.yml]: fixed

File: .pre-commit-config.yaml (hunk 2 above)
Comment (yunchu) [reply .pre-commit-config.yaml:44]: can you add it?

Comment (vinnamkim) [reply .pre-commit-config.yaml:44]: OK

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Pull request merged**